exasol/ds/sandbox/lib/dss_docker/create_image.py

```diff
@@ -0,0 +1,99 @@
+import docker
+import logging
+from datetime import datetime
+from docker.types import Mount
+from exasol.ds.sandbox.lib import pretty_print
+from importlib_metadata import version
+from pathlib import Path
+
+from exasol.ds.sandbox.lib.config import ConfigObject, SLC_VERSION
+# renaming proposal:
+# import exasol.ds.sandbox.lib.ansible
+# file ansible.__init__.py with content "import ...ansible.repository"
+# enables usage: ansible.repository.default
+from exasol.ds.sandbox.lib.ansible import ansible_repository
+from exasol.ds.sandbox.lib.ansible.ansible_run_context import AnsibleRunContext
+from exasol.ds.sandbox.lib.ansible.ansible_access import AnsibleAccess
+from exasol.ds.sandbox.lib.setup_ec2.run_install_dependencies import run_install_dependencies
+
+
+DSS_VERSION = version("exasol-data-science-sandbox")
+CONTAINER_NAME = "ds-sandbox-docker"
+
+
+class DssDockerImage:
+    def __init__(
+            self,
+            repository: str,
+            version: str = None,
+            publish: bool = False,
+            log_level: str = logging.INFO,
+    ):
+        version = version if version else DSS_VERSION
+        self.container_name = CONTAINER_NAME
+        self.image_name = f"{repository}:{version}"
+        self.publish = publish
+        self.log_level = log_level
+
+    def _ansible_run_context(self) -> AnsibleRunContext:
+        extra_vars = {
+            "docker_container": self.container_name,
+        }
+        return AnsibleRunContext(
+            playbook="dss_docker_playbook.yml",
+            extra_vars=extra_vars,
+        )
+
+    def _ansible_config(self) -> ConfigObject:
+        return ConfigObject(
+            time_to_wait_for_polling=0.1,
+            slc_version=SLC_VERSION,
+        )
+
+    def create(self):
+        logger = logging.getLogger(__name__)
+        logger.setLevel(self.log_level)
+        logging.basicConfig(format="%(asctime)s %(levelname)s %(message)s")
+        try:
+            start = datetime.now()
+            docker_client = docker.from_env()
+            path = Path(__file__).parent
+            logger.info(
+                f"Creating docker image {self.image_name}"
+                f" from {path / 'Dockerfile'}"
+            )
+            docker_client.images.build(path=str(path), tag=self.image_name)
+            socket_mount = Mount("/var/run/docker.sock", "/var/run/docker.sock", type="bind")
```

Comment:
Mount shouldn't be needed during build

Reply:
However, you might want to add it to the dockerfile

https://docs.docker.com/engine/reference/builder/#volume